### PR TITLE
Add Gemma4 layer-wise unit tests

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Add layer-wise unit tests comparing MaxText and PyTorch implementations for Gemma 4 vision components, including `VisionEntry`, `Gemma4VisionRotaryEmbedding`, `Gemma4(Vision)Attention`, `Gemma4EncoderBlock`, `VisionExit`, `Gemma4VisionEncoderLayer`, and `Gemma4VisionProjector` (**still offline**).

**TODO**: A follow-up PR to transform such layer-wise tests to PyTorch-free and runnable on CI.

# Tests

```
python -m pytest tests/unit/gemma4_layers_test.py -vv -s
```

```
collected 7 items                                               

tests/unit/gemma4_layers_test.py::TestGemma4VisionEntry::test_vision_entry_matches_torch W0514 05:57:15.485060 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
tests/unit/gemma4_layers_test.py::TestGemma4VisionRotaryEmbedding::test_rotary_embedding_matches_torch W0514 05:57:20.120236 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
tests/unit/gemma4_layers_test.py::TestGemma4VisionAttention::test_attention_matches_torch W0514 05:57:23.553000 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
tests/unit/gemma4_layers_test.py::TestGemma4VisionEncoderBlock::test_encoder_block_matches_torch W0514 05:57:29.738181 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
tests/unit/gemma4_layers_test.py::TestGemma4VisionExit::test_vision_exit_matches_torch W0514 05:57:31.237544 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
tests/unit/gemma4_layers_test.py::TestGemma4VisionEncoderEndToEnd::test_vision_encoder_matches_torch W0514 05:57:44.092117 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
tests/unit/gemma4_layers_test.py::TestGemma4VisionProjector::test_vision_projector_matches_torch W0514 05:57:45.259719 2196772 pjrt_executable.cc:642] Assume version compatibility. PjRt-IFRT does not track XLA executable versions.
PASSED
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/unit/gemma4_layers_test.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+  np.random.seed(42)
+  torch.manual_seed(42)
+
+
+# =============================================================================
+# Weight Copying Helpers
+# =============================================================================
+
+
+def copy_rmsnorm_weights(torch_norm, jax_norm):
```
Comment (aireenmei) [inline tests/unit/gemma4_layers_test.py]: Seems not very specific to gemma4? Should we move to tests.utils.multimodal_test_utils for re-use?

File: tests/unit/gemma4_layers_test.py [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+
+  def test_attention_matches_torch(self):
+    torch_model = TorchGemma4VisionAttention(torch_vision_config, layer_idx=0)
+    torch_model.eval()
+
+    batch_size = 2
+    seq_len = 42 * 60
+    dummy_shape = (batch_size, seq_len, self.config.hidden_size_for_vit)
+
+    jax_model = JaxGemma4Attention(
```
Comment (aireenmei) [inline tests/unit/gemma4_layers_test.py]: How about JaxGemma4VisionAttention?

**Review by aireenmei — APPROVED** [review]
Thanks for adding tests!

File: tests/unit/gemma4_layers_test.py [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+    attention_type="full",
+    matmul_precision="highest",
+    dropout_rate=0.0,
+    dtype="float32",
+    dtype_mm="float32",
+    weight_dtype="float32",
+    float32_logits=True,
+    float32_qk_product=True,
+)
+
```
[the remark this answers is not in the document]
Comment (hengtaoguo) [reply tests/unit/gemma4_layers_test.py:78]: Removed unused gemma4_config_path var.

File: tests/unit/gemma4_layers_test.py (hunk 1 above)
Comment (hengtaoguo) [reply tests/unit/gemma4_layers_test.py]: Good call! Moved `copy_rmsnorm_weights ` to `multimodal_test_utils`.

File: tests/unit/gemma4_layers_test.py (hunk 2 above)
Comment (hengtaoguo) [reply tests/unit/gemma4_layers_test.py]: Just renamed, thanks!

**Review by NuojCheng — APPROVED** [review]
